### PR TITLE
Use Selectorbar for Sample Code

### PR DESCRIPTION
**Pull request by ghost1372**
related to #1535 

Full Code (Xaml/C#)

Single Code (Xaml Only)

## How Has This Been Tested?
Tested on my Laptop
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by Jay-o-Way** [comment]
Does this solve #704?

**Comment by ghost1372** [comment]
> Does this solve #704?

No.  #704 needs more investigation.

**Review by marcelwgn — COMMENTED** [review]
Thank you for creating this PR! 

I think porting the SwitchPresenter control over just for this is a bit overkill, this problem can also be solved with a bit of code-behind. Also, given that people also use this app's source code for reference, introducing a new control just for this doesn't seem like the best idea.

**Comment by ghost1372** [comment]
> Thank you for creating this PR! 
> 
> I think porting the SwitchPresenter control over just for this is a bit overkill, this problem can also be solved with a bit of code-behind. Also, given that people also use this app's source code for reference, introducing a new control just for this doesn't seem like the best idea.

I saw that the gallery uses some communitytoolkit Nuget packages.
For a simple control, I did not use the community Nuget package, which has a larger size... 
That's why I copied their codes.

**Comment by marcelwgn** [comment]
We are in fact using Community Toolkit packages, that is correct. I would argue though, in this case, porting a complete control for something one could do with a few lines of code behind is overkill.

**Comment by ghost1372** [comment]
> We are in fact using Community Toolkit packages, that is correct. I would argue though, in this case, porting a complete control for something one could do with a few lines of code behind is overkill.

ok, what do you suggest? Changing visibility? Frame?

**Comment by marcelwgn** [comment]
> ok, what do you suggest? Changing visibility? Frame?

I would probably use a ContentPresenter and have that switch the content

**Comment by ghost1372** [comment]
> > ok, what do you suggest? Changing visibility? Frame?
> 
> I would probably use a ContentPresenter and have that switch the content

You said late, sorry, please check the last commit

**Comment by ghost1372** [comment]
@marcelwgn I saw your last message late, I sent 2 commits, First by changing the visibility and lastly the suggestion you gave.

**Comment by Jay-o-Way** [comment]
Looks like the control is visible, even if content is empty.
image

**Comment by ghost1372** [comment]
> For some reason - for some controls, the content is empty. image image

hmm... it seems that XamlSource is just a file path, so content will be updated later after reading file.

**Comment by ghost1372** [comment]
@Jay-o-Way  i reverted last commit (ContentPresenter) now everything is working fine

File: WinUIGallery/Controls/ControlExample.xaml [hunk 1]
```diff
@@ -138,26 +160,36 @@
                 <muxc:Expander.Header>
                     <TextBlock Text="Source code" />
                 </muxc:Expander.Header>
+                <Grid>
```
Comment (Jay-o-Way) [inline WinUIGallery/Controls/ControlExample.xaml:163]:
```suggestion
                <Grid RowSpacing="8">
```
If it were me, I would add just a few pixels between the SelectorBar and the Grid.

**Review by Jay-o-Way — COMMENTED** [review]
Nit: every **first** expanding event makes the content move twice, very quickly.

**Comment by Jay-o-Way** [comment]
> > Does this solve #704?
> 
> No. we need to disable horizontal scroll

For me, this pr currently actually _does_ disable horizontal movement when using the scroll wheel hovering this area. 🥳 @karkarl

**Comment by karkarl** [comment]
> > > Does this solve #704?
> > 
> > 
> > No. we need to disable horizontal scroll
> 
> For me, this pr currently actually _does_ disable horizontal movement when using the scroll wheel hovering this area. 🥳 @karkarl

I'm ok disabling horizontal scroll. It annoys me a lot too :P

**Comment by Jay-o-Way** [comment]
Question: I see `muxc:Expander` is used, but do we still need the `muxc` part?
One more detail please? Removing this margin/padding would be pretty.
image

File: WinUIGallery/Controls/ControlExample.xaml [hunk 2]
```diff
@@ -138,26 +160,38 @@
                 <muxc:Expander.Header>
                     <TextBlock Text="Source code" />
                 </muxc:Expander.Header>
+                <Grid RowSpacing="8">
```
Comment (niels9001) [inline WinUIGallery/Controls/ControlExample.xaml:163]: Maybe this could be 16, to provide a bit more whitespace?

Comment (ghost1372) [reply WinUIGallery/Controls/ControlExample.xaml]: done

**Comment by ghost1372** [comment]
> > > > Does this solve #704?
> > > 
> > > 
> > > No. we need to disable horizontal scroll
> > 
> > 
> > For me, this pr currently actually _does_ disable horizontal movement when using the scroll wheel hovering this area. 🥳 @karkarl
> 
> I'm ok disabling horizontal scroll. It annoys me a lot too :P

done

**Comment by Jay-o-Way** [comment]
Hold on, I see a little misunderstanding here. The issue in 704 was: When scrolling (vertically) across a page, coming over a code example gives an unwanted horizontal scrolling. This PR already solved that. The last commit makes it **impossible** to scroll horizontally. Please revert.

**Review by Jay-o-Way — CHANGES_REQUESTED** [review]
Hold on, I see a little misunderstanding here. The issue in 704 was: When scrolling (vertically) across a page, coming over a code example gives an unwanted horizontal scrolling. This PR already solved that. The last commit makes it **impossible** to scroll horizontally. Please revert.

**Comment by ghost1372** [comment]
> Hold on, I see a little misunderstanding here. The issue in 704 was: When scrolling (vertically) across a page, coming over a code example gives an unwanted horizontal scrolling. This PR already solved that. The last commit makes it **impossible** to scroll horizontally. Please revert.

Done

**Review by Jay-o-Way — APPROVED** [review]

**Comment by ghost1372** [comment]
> I think we'd want to horizontally align the XAML / C# SelectorBar with the XAML?

Done
image

**Review by niels9001 — APPROVED** [review]
LGTM! Thanks!!

**Pull request merged**